examples/2d/contributors.rs

```diff
@@ -163,40 +165,34 @@ fn select_system(
     mut query: Query<(&Contributor, &mut Sprite, &mut Transform)>,
     time: Res<Time>,
 ) {
-    let mut timer_fired = false;
-    for mut timer in timer_query.iter_mut() {
-        if !timer.tick(time.delta()).just_finished() {
-            continue;
-        }
-        timer.reset();
-        timer_fired = true;
-    }
-
-    if !timer_fired {
+    let mut timer = timer_query.single_mut();
+    if !timer.tick(time.delta()).just_finished() {
         return;
     }
-
-    let prev = contributor_selection.idx;
-
-    if (contributor_selection.idx + 1) < contributor_selection.order.len() {
-        contributor_selection.idx += 1;
-    } else {
-        contributor_selection.idx = 0;
+    if timer.times_finished() == 1 {
```

Comment:
Was your intention here to only update `text.sections[0]` once, rather than every time a new contributor is selected?

If so, `timer_finished()` resets after `.tick()` so I don't think this is doing what you want.

Reply:
Dang, you're right. I used a local to fix this, please let me know if there's a more idiomatic way of doing this.